src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+    private T[] arrayList;
+    private int size = 0;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void resizeArray() {
+        if (arrayList.length == size) {
```

Comment:
You have this IF inside the` resizeArray()` method. So, sometimes you resize, sometimes not. You should remove this IF and make resize it every time (call this method inside IF) **or** rename this method to make clear how it works (resizeIfArrayFull() or something like this).